hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1163,6 +1166,132 @@ private class Emit(
             res.isNull))
 
         EmitTriplet(setup, m, res.invoke[Double]("doubleValue"))
+      case x@CollectDistributedArray(contexts, globals, cname, gname, body) =>
+        val ctxType = coerce[TArray](contexts.typ).elementType.physicalType
+        val gType = globals.typ.physicalType
+        val bType = body.typ.physicalType
+
+        val ctxTypeTuple = PTuple(FastIndexedSeq(ctxType))
+        val gTypeTuple = PTuple(FastIndexedSeq(gType))
+        val bTypeTuple = PTuple(FastIndexedSeq(bType))
```

Comment:
I don't quite understand why these have to be wrapped in tuples

Reply:
Wrapping things in tuples is the easiest way to serialize missingness.